config/initializers/apipie.rb

```diff
@@ -113,3 +113,35 @@ def description
         "dot(.), space, underscore(_), hypen(-) with no leading or trailing space."
   end
 end
+
+# Allows to enumerate multiple types that a parameter accepts.
+class AnyOfTypesValidator < Apipie::Validator::BaseValidator
+  def initialize(param_description, argument, options = {})
+    super(param_description)
+    @allowed_types = options[:of] || []
+  end
+
+  def validate(value)
+    # The validator has rather informative value, skip the real validation
+    true
+  end
+
+  def self.build(param_description, argument, options, block)
+    if argument == :any_type
+      self.new(param_description, argument, options)
+    end
+  end
+
+  def description
+    if @allowed_types.empty?
+      'Can be any type'
+    else
+      types = @allowed_types.map { |type| "<code>#{type}</code>" }.join(', ')
+      'Must be one of types: %s' % types.join(', ')
+    end
+  end
+
+  def expected_type
```

Comment:
Is this method used anywhere?

Reply:
Apipie exposes this in json doc exports and hammer uses it together with a validator description to determine option type. I don't think it's exposed in apipie's web UI though.

E.g. when enum is used:
```
[1] pry(#<HammerCLI::Apipie::OptionBuilder>)> param.validator
=> "Must be one of: string, boolean, integer, real, array, hash, yaml, json." 
[2] pry(#<HammerCLI::Apipie::OptionBuilder>)> param.expected_type.to_s
=> "string" 
```

and when one of types validator is used:
```
[1] pry(#<HammerCLI::Apipie::OptionBuilder>)> param.validator
=> "Must be one of types: string, boolean, integer, real, array, hash, yaml, json." 
[2] pry(#<HammerCLI::Apipie::OptionBuilder>)> param.expected_type.to_s
=> "any_of_enumerated" 
```

Reply:
Ah, ok. 

My last comment/question is about consistency: we have `AnyOfTypesValidator`, `:any_type` and `:any_of_enumerated` in the code, naming the same thing at different places. I wonder if unifying on `AnyTypeValidator` + `:any_type` here and on other places wouldn't help the consistency.

Reply:
Makes sense, updated.